src/Servers/Kestrel/test/InMemory.FunctionalTests/ResponseHeaderTests.cs

```diff
@@ -100,4 +101,42 @@ public async Task ResponseHeaders_WithInvalidValuesAndCustomEncoder_AbortsConnec
 
         await connection.ReceiveEnd();
     }
+
+    [Fact]
+    public async Task ResponseHeaders_NullEntriesAreIgnored()
+    {
+        string tag = "tag";
```

Comment:
```suggestion
        var tag = "tag";
```

Reply:
How did you choose the header name `tag` for your test? The original repro used `warning`.

While the headers collection looks like a dictionary, there's a custom implementation inside that special cases common headers:
https://github.com/dotnet/aspnetcore/blob/39219394f66854db5d556ad78f9bb065ee48b585/src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs#L3747-L3752

`warning` is one of the special headers and `tag` is not, so there might be different behavior.

Reply:
I tried both in my testing, but I can change this to be `"warning"` again.